tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/cocosnet_evaluator.py

```diff
@@ -0,0 +1,483 @@
+
+    def process_dataset(
+            self, subset=None,
+            num_images=None,
+            check_progress=False,
+            dataset_tag='',
+            output_callback=None,
+            allow_pairwise_subset=False,
+            dump_prediction_to_annotgiation=False,
+            **kwargs):
+        if self.dataset is None or (dataset_tag and self.dataset.tag != dataset_tag):
+            self.select_dataset(dataset_tag)
+
+        self._annotations, self._predictions = [], []
+
+        self._create_subset(subset, num_images, allow_pairwise_subset)
+
+        if 'progress_reporter' in kwargs:
+            _progress_reporter = kwargs['progress_reporter']
+            _progress_reporter.reset(self.dataset.size)
+        else:
+            _progress_reporter = None if not check_progress else self._create_progress_reporter(
+                check_progress, self.dataset.size
+            )
+
+        metric_config = self.configure_intermediate_metrics_results(kwargs)
+        compute_intermediate_metric_res, metric_interval, ignore_results_formatting = metric_config
+
+        for batch_id, (batch_input_ids, batch_annotation, batch_inputs, batch_identifiers) in enumerate(self.dataset):
+            batch_inputs = self._preprocessing_for_batch_input(batch_annotation, batch_inputs)
+            extr_batch_inputs, _ = extract_image_representations(batch_inputs)
+            batch_predictions = self.test_model.predict(batch_identifiers, extr_batch_inputs)
+            annotations, predictions = self.postprocessor.process_batch(batch_annotation, batch_predictions)
+
+            metrics_result, _ = self.metric_executor.update_metrics_on_batch(batch_input_ids, annotations, predictions)
+            gan_annotations = []
+            gan_predictions = []
+            if self.check_model:
+                for index_of_metric in range(self.check_model.number_of_metrics):
+                    gan_annotations.extend(self.check_model.predict(batch_identifiers, annotations, index_of_metric))
+                    gan_predictions.extend(self.check_model.predict(batch_identifiers, predictions, index_of_metric))
+                batch_identifiers.extend(batch_identifiers)
+                gan_annotations = [RawTensorAnnotation(batch_identifier, item)
+                                   for batch_identifier, item in zip(batch_identifiers, gan_annotations)]
+                gan_predictions = [RawTensorPrediction(batch_identifier, item)
+                                   for batch_identifier, item in zip(batch_identifiers, gan_predictions)]
+
+            if output_callback:
+                output_callback(
+                    predictions,
+                    metrics_result=metrics_result,
+                    element_identifiers=batch_identifiers,
+                    dataset_indices=batch_input_ids
+                )
+
+            if self.metric_executor.need_store_predictions:
+                self._annotations.extend(gan_annotations)
```

Comment:
probably need to save separated list of metrics and annotations/predictions with usage verification network and without it (in both cases you can have metric s=which required save predictions)